demo/README.md

```diff
@@ -0,0 +1,225 @@
+
+## 3 Detailed Tools Configurations
+
+### 3.1 Configuring kern_bin_db and the local database
+
+ks-nav is comprised of three different tools. The first tool to run  **kern_bin_db**. It acquires data from a kernel image and builds a structured kernel symbol and cross-reference database in SQL format. 
+In this section, we dive into the configuration of **kern_bin_db** for this demo. 
+
+The kern_bin_db main source code is in *ks-nav/kern_bin_db/main.go*. Once successfully built, it generates the ***nav-db-filler*** executable. 
+The Makefile under folder *ks-nav/kern_bin_db/* provides other build options. Once the executable is generated, the tool requires the following inputs to run successfully: 
+- a json configuration file to run (we'll use file '*./demo/ksnav_app_linux/ksnav_kdb.json*' for this example)
+- a Linux image with debug symbols (located in the /app folder in the demo container)
+
+File *ksnav_kdb.json* is shown below: 
+```bash
+{
+"LinuxWDebug":		"/app/vmlinux", 
+"LinuxWODebug":		"/app/vmlinux.work",
+"StripBin":		"/usr/bin/strip",
+"DBDriver":		"postgres",
+"DBDSN":		"host=localhost port=5432 user=postgres password=my123 dbname=kernel_bin sslmode=disable",
+"Maintainers_fn":	"/app/MAINTAINERS",
+"KConfig_fn":		"include/generated/autoconf.h",
+"KMakefile":		"/app/Makefile",
+"Mode":			15,
+"Note":			"upstream"
+}LinuxWDebug: The path to the linux binary unstripped of debug symbols
+
+
+```
+- *LinuxWDebug*: The path to the linux binary unstripped of debug symbols
+- *LinuxWODebug*: The path to the Linux binary once stripped of debug symbols. In this example, the default strip command in the container is used. When using a cross-compiled image, the appropriate strip library must be used. The path to the strip command to use is defined in the next config parameter
+- *StripBin*: The path to the strip command used to generate 'LinuxWODebug'
+- *DBDriver*: The specified Database to use (can be either postgres, mysql or sqlite3)
+- *DBDSN*: The details required to connect to the database. In this demo, the database is local to the container. 
+- *Maintainers_fn*: ks-nav can organize diagrams by sub-systems. The Maintainers file is parsed for that purpose. This config param is the path to Maintainers file. 
+- *KConfig_fn*: The path to the build configuration 
+- *KMakefile*: This should point to the Makefile
+- *Mode*: This is a mode of operation for debug purposes - linked to the database installed (which in this case is postgres:15). Should be defaulted to 15 in this demo. 
+- *Note*: This is the string copied to the database - a way to tag the dataset
+
+In Yocto environments, the Linux source and Linux build folders usually tend to be different paths. The user should take care to point to the appropriate path. 
+
+File *'./demo/ksnav_app_linux/postgres_conf_template.conf'* is the file used to configure the database in the container. It is copied into the container's /tmp folder. 
```

Comment:
Can use use the original template?   https://github.com/elisa-tech/ks-nav/blob/main/container/postgres_conf_template.conf

Reply:
use original with typo fix

Reply:
integrated in latest commits